code/modules/mob/living/simple_animal/hostile/roguetown/jitterskull.dm

```diff
@@ -0,0 +1,1349 @@
+    armor_penetration = 100
+    // It attacks anything and everything
+    attack_same = TRUE
+    attack_sound = 'sound/combat/wooshes/bladed/wooshmed (1).ogg'
+    dodge_sound = 'sound/combat/dodge.ogg'
+    parry_sound = "bladedmedium"
+    d_intent = INTENT_PARRY
+    speak_emote = list("chatters")
+    del_on_death = TRUE
+    STALUC = 10
+    STASTR = 20
+    STACON = 20
+    STASPD = 18
+    atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+    minbodytemp = 0
+    // Empty faction so basic targeting treats everything as valid
+    faction = list()
+    footstep_type = null
+    defprob = 40
+    canparry = TRUE
+    retreat_health = null
+    can_have_ai = FALSE // disable native simple_animal AI; we use an ai_controller
+    AIStatus = AI_OFF
+    ai_controller = /datum/ai_controller/haunt
+
+    // Behavior control
+    var/next_attack_time = 0       // world.time until which we will refrain from attacking
+    var/teleport_cooldown_until = 0 // throttle for on-hit teleport reactions
+    var/next_idle_chatter_time = 0  // next time we're allowed to idle-chatter
+    var/taunt_cooldown_until = 0    // cooldown for taunting after a down
+    var/is_dying = FALSE            // guard to prevent re-entrant death sequences
+    var/mob/living/reengage_target  // who we plan to reengage after a panic blink
+    var/original_alpha = 255
+    var/suppress_panic_until = 0      // during/after a strike, ignore panic blink to avoid double-teleport
+    var/attack_pose_until = 0         // world.time until which we should remain in attack pose
+    var/attack_swing_until = 0        // window covering the active swing where we might ignore counter-hits
+    var/ignore_hits_this_swing = FALSE // 50% chance: during attack swing, ignore incoming strikes (no pain/teleport)
+    var/next_grab_counter_time = 0    // cooldown before we do another anti-grab counter
+    // Critical finisher throttle
+    var/next_critical_available_time = 0 // world.time when we can attempt a critical finisher again
+    // Near-instant attack cadence
+    melee_cooldown = 0.1 SECONDS
+    var/is_feasting = FALSE         // currently lingering and consuming an unconscious target
+    // Anti-stuck tracking
+    var/last_stuck_x = 0
+    var/last_stuck_y = 0
+    var/stuck_ticks = 0
+    // Stalking and aggression flags
+    var/is_stalking = FALSE        // briefly TRUE when we blink away and "stare" before re-engaging
+    var/angry_until = 0            // during this window, skip whiffs and any artificial waiting
+    // Vendetta focus: keep attention on whoever last hurt us for a while
+    var/mob/living/vendetta_target = null
+    var/vendetta_until = 0
+    // Tether throttle to avoid spam teleports
+    var/next_tether_allowed = 0
+    // Guarding: linger around a helpless target before feasting
+    var/is_guarding = FALSE
+    var/mob/living/guarding_target = null
```

Comment:
potential harddel source, this should 1) be handled on an AI controller/AI behavior datum, and 2) use a weakref to avoid harddels